src/main/java/com/yahoo/memory/AllocateDirect.java

```diff
@@ -26,40 +26,39 @@
    * <p>Allocates and provides access to capacityBytes directly in native (off-heap) memory
    * leveraging the Memory interface.
    * The allocated memory will be 8-byte aligned, but may not be page aligned.
-   * @param capacity the the requested capacity of off-heap memory. Cannot be zero.
    * @param state contains valid, capacity at this point
    */
-  private AllocateDirect(final long capacity, final ResourceState state) {
+  private AllocateDirect(final ResourceState state) {
```

Comment:
The reason I added capacity to this method was because I am trying to remove dependencies on ResourceState.  So that, eventually RS can be removed.  I haven't figured out how to do the entire removal in one step, so this was an intermediate step along the way.  

It was also my understanding that you also want to eventually get rid of RS.  Have you changed your mind?

Reply:
Please revert.

Reply:
I essentially reverted back to my original code, so I can move forward in simplifying the `ResourceState`.